x-pack/elastic-agent/pkg/artifact/download/composed/downloader_test.go

```diff
@@ -58,7 +59,9 @@ func TestComposed(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		d := NewDownloader(tc.downloaders[0], tc.downloaders[1])
+		log, _ := logger.New("", false)
+
+		d := NewDownloader(log, tc.downloaders[0], tc.downloaders[1])
```

Comment:
Why not:
```suggestion
		d := NewDownloader(log,  tc.downloaders...)
```
?

Reply:
Oh I didn't check the arguments, just added a log to have some debug log lines for the composite downloader, I will have another look at the test

Reply:
The test requires a `[]CheckableDownloader` because of the check function while the `NewDownloader` calls for a `[]download.Downloader` which is a different from `[]CheckableDownloader` so I guess I will leave it as it is